podman/domain/system.py

```diff
@@ -51,11 +50,23 @@ def login(
             email: Registry account email address
             registry: URL for registry access. For example,
                 https://quay.io/v2
-            reauth: If True, refresh existing authentication. Default: False
-            dockercfg_path: Path to custom configuration file.
-                Default: $HOME/.config/containers/config.json
         """
 
+        payload = {
+            "username": username,
+            "password": password,
+            "email": email,
+            "serveraddress": registry,
+        }
```

Comment:
```suggestion
        }
        payload = api.prepare_body(payload)
```

will clean up any field that are None from caller.